theatre/studio/src/store/stateEditors.ts

```diff
@@ -245,6 +245,42 @@ namespace stateEditors {
                   ])
                 }
               }
+
+              export function replaceMarker(
+                p: SheetAddress,
+                marker: {time: number},
+              ) {
+                const sequenceEditor =
+                  stateEditors.studio.historic.projects.stateByProjectId.stateBySheetId._ensure(
+                    p,
+                  ).sequenceEditor
+
+                if (!sequenceEditor.markers) {
+                  sequenceEditor.markers = [marker]
+                } else {
+                  sequenceEditor.markers = [
+                    ...sequenceEditor.markers.filter(
+                      (otherMarker) => otherMarker.time !== marker.time,
+                    ),
+                    marker,
+                  ]
+                }
+              }
+
+              export function removeMarker(p: SheetAddress, atTime: number) {
+                const sequenceEditor =
+                  stateEditors.studio.historic.projects.stateByProjectId.stateBySheetId._ensure(
+                    p,
+                  ).sequenceEditor
```

Comment:
Nit: `p` is kinda weird shortened identifier

Reply:
We renamed it to `options`